lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -755,62 +828,77 @@
-                        var key = parts[1];
-                        var value = parts[2] === "" ? undefined : parts[2];
+                    for (let i = 0; i < getArrayLength(resolvedLocaleInfo.subTags); i += 2) {
+                        const key = resolvedLocaleInfo.subTags[i];
+                        const value = resolvedLocaleInfo.subTags[i + 1] || undefined; // normalize falsy values to undefined
                         if (key === "kf" && caseFirst === undefined) {
-                            caseFirst = GetOption(setPrototype({ caseFirst: value }, null), "caseFirst", "string", ["upper", "lower", "false"], undefined);
+                            caseFirst = GetOption(bare({ caseFirst: value }), "caseFirst", "string", ["upper", "lower", "false"], undefined);
                         } else if (key === "kn" && numeric === undefined) {
                             if (value !== undefined) {
-                                numeric = Internal.ToLogicalBoolean(callInstanceFunc(StringInstanceToLowerCase, value) === "true");
+                                numeric = Internal.ToLogicalBoolean(_.toLowerCase(value) === "true");
                             } else {
                                 numeric = true;
                             }
                         } else if (key === "co" && value !== undefined && value !== "default" && value !== "search" && value !== "sort" && value !== "standard") {
                             // Ignore these collation values as they shouldn't have any impact
                             collation = value;
                         }
-                    });
+                    }
                 }
-                if (collation !== "default") {
-                    var accepedCollationForLocale = localesAcceptingCollationValues[collationAugmentedLocale];
-                    var windowsCollation = "";
-                    if (accepedCollationForLocale !== undefined && (windowsCollation = accepedCollationForLocale[collation]) !== undefined) {
-                        if (windowsCollation !== "") {
-                            collationAugmentedLocale = collationAugmentedLocale + "_" + windowsCollation;
-                        }
+
+                // re-append the collation to the base locale string, if we have found one.
+                // this is going to be the final locale that we report back to the user in resolvedOptions.
+                // even if the user provided -kn or -kf tags, NLS doesnt support them, and its nice to have a common denominator of everyone understanding language + script + region + collation.
+                // firefox (~57) keeps -kn, -kf, and -co in the returned locale (if provided), chrome (~61) will only keep -co
+                let finalLocale; // at most a langauge, script, region, and collation value
+                let platformLocale; // finalLocale as BCP47 for ICU, RFC4646 for NLS
```

Comment:
>BCP47 [](start = 54, length = 5)

to be specific RFC5646 (RFC4646 was a part of BCP47 before RFC5646 obsoleted it).